chromium_src/third_party/blink/renderer/core/exported/web_view_impl.cc

```diff
@@ -5,18 +5,28 @@
 
 #include "third_party/blink/renderer/core/exported/web_view_impl.h"
 
+#include "third_party/blink/public/common/web_preferences/web_preferences.h"
 #include "third_party/blink/public/platform/web_runtime_features.h"
+#include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 
-#if BUILDFLAG(IS_MAC)
-#define SetAccelerated2dCanvasEnabled(...)             \
-  SetAccelerated2dCanvasEnabled(__VA_ARGS__);          \
-  if (prefs.disable_web_share) {                       \
-    RuntimeEnabledFeatures::SetWebShareEnabled(false); \
+// not a funtions because Set* methods are protected and WebView is a friend of
+// RuntimeEnabledFeatures.
+#define BRAVE_APPLY_WEB_PREFERENCES(prefs)                 \
```

Comment:
Can't this be a global function instead of a macro?
```cpp
void ApplyBraveWebPreferences(const web_pref::WebPreferences& prefs) {
  RuntimeEnabledFeatures::SetBraveIsInTorContextEnabled(
        prefs.is_tor_window);
    if constexpr (BUILDFLAG(IS_MAC)) {
      if (prefs.is_tor_window) {
        RuntimeEnabledFeatures::SetWebShareEnabled(false);
      }
    }
}
```

Reply:
No It can't, according to the comment for this define.